core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -179,11 +186,9 @@ public ChannelImpl shutdown() {
       savedActiveTransport = activeTransport;
       if (savedActiveTransport != null) {
         activeTransport = null;
-      } else if (transports.isEmpty()) {
-        terminated = true;
-        lock.notifyAll();
-        onChannelTerminated();
       }
+      // TODO(carl-mastrangelo): maybe remove this outside of the sync
+      checkTerminated();
```

Comment:
Still keep this in an else? We know it won't do anything if the active transport is not shut down.